### ISSUE TITLE
Platform-specific function ABIs are not validated

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust#81396]
This code builds (and runs) fine on stable Rust when targeting x86_64-unknown-linux-gnu:

```rust
extern "aapcs" fn f() {}

fn main() {
    f();
}
```

Other nonsensical ABIs are also accepted, including `win64`, `avr-interrupt` (with feature gate). I expected all of these to result in a compilation error, since they are unsupported on x86_64-unknown-linux-gnu.

**Comment by hyd-dev** [comment]
cc #57182 #86231

**Comment by nagisa** [comment]
Duplicate of #57182
